main.go

```diff
@@ -0,0 +1,229 @@
+	}
+
+	if releaseArch.Media.Qemu != nil {
+		qemu := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Qemu.Artifacts,
+		}
+		artifacts.Qemu = &qemu
+	}
+
+	if releaseArch.Media.Virtualbox != nil {
+		virtualbox := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Virtualbox.Artifacts,
+		}
+		artifacts.Virtualbox = &virtualbox
+	}
+
+	if releaseArch.Media.Vmware != nil {
+		vmware := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Vmware.Artifacts,
+		}
+		artifacts.Vmware = &vmware
+	}
+
+	if releaseArch.Media.Metal != nil {
+		metal := StreamMediaDetails{
+			Release: release.Release,
+			Formats: releaseArch.Media.Metal.Artifacts,
+		}
+		artifacts.Metal = &metal
+	}
+
+	streamArch := StreamArch{
+		Artifacts: artifacts,
+		Updates:   StreamUpdates{Release: release.Release},
+		Images:    &cloudImages,
+	}
+	return streamArch
+}
+
+var errReleaseIndexMissing = errors.New("Please specify release index url")
+
+func run() error {
+	var releasesURL string
+
+	flag.StringVar(&releasesURL, "release-index", "", "Release index location for the required stream")
+	var overrideFilename string
+	flag.StringVar(&overrideFilename, "override", "", "Override file location for the required stream")
+
+	flag.Parse()
+
+	if releasesURL == "" {
+		return errReleaseIndexMissing
+	}
+
+	releasePath, err := ReleaseURL(releasesURL)
+
+	if err != nil {
+		return fmt.Errorf("Error with Release Index: %v", err)
+	}
+
+	var release Release
+	resp, err := http.Get(releasePath)
+	if err != nil {
+		return fmt.Errorf("Error while fetching: %v", err)
+	}
+
+	defer resp.Body.Close()
+
+	decoder := json.NewDecoder(resp.Body)
+	if err := decoder.Decode(&release); err != nil {
+		return fmt.Errorf("Error while decoding json: %v", err)
+	}
+
+	streamArch := make(map[string]*StreamArch)
+	for arch, releaseArch := range release.Architectures {
+		archContent := releaseToStream(releaseArch, release)
+		streamArch[arch] = &archContent
+	}
+
+	streamMetadata := StreamMetadata{
+		Stream:        release.Stream,
+		Metadata:      Metadata{LastModified: time.Now().UTC().Format(time.RFC3339)},
+		Architectures: streamArch,
+	}
+
+	file, err := os.Create("stream.json")
+	if err != nil {
+		return fmt.Errorf("Can't Open file: %v", err)
+	}
+
+	defer file.Close()
+
+	if overrideFilename != "" {
+		overrideFile, err := os.Open(overrideFilename)
+		if err != nil {
+			return fmt.Errorf("Can't open file %s: %v", overrideFilename, err)
+		}
+		defer overrideFile.Close()
+
+		overrideDecoder := json.NewDecoder(overrideFile)
+		err = overrideDecoder.Decode(&streamMetadata)
+		if err != nil {
+			return fmt.Errorf("Error while decoding: %v", err)
+		}
+	}
+
+	encoder := json.NewEncoder(file)
+	err = encoder.Encode(&streamMetadata)
+	if err != nil {
+		return fmt.Errorf("Error while encoding: %v", err)
+	}
+	return nil
+
+}
+
+func main() {
+	err := run()
+	if err != nil {
```

Comment:
Minor style suggestion: you can write this as `if err := run(); err != nil {...}`.

Reply:
I am still picking up Go, thanks for the suggestion, fixed!